coresimd/src/x86/i586/sse.rs

```diff
@@ -1572,7 +1571,7 @@ pub const _MM_HINT_NTA: i8 = 0;
 #[cfg_attr(test, assert_instr(prefetcht1, strategy = _MM_HINT_T1))]
 #[cfg_attr(test, assert_instr(prefetcht2, strategy = _MM_HINT_T2))]
 #[cfg_attr(test, assert_instr(prefetchnta, strategy = _MM_HINT_NTA))]
-pub unsafe fn _mm_prefetch(p: *const c_void, strategy: i8) {
+pub unsafe fn _mm_prefetch(p: *const u8, strategy: i8) {
```

Comment:
I don't know why this intrinsic used `c_void`, but [Intel specifies it as taking a `const char*`]( https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=prefetch&expand=4101). 

The intrinsic is also untested, I've filled: https://github.com/rust-lang-nursery/stdsimd/issues/198